h/static/scripts/tests/controllers/autosuggest-dropdown-controller-test.js

```diff
@@ -0,0 +1,413 @@
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      syn
+        .click(input)
+        .type('t', () => {
+
+          const items = getListItems();
+
+          assert.lengthOf(items, 1);
+
+          assert.isTrue(isSuggestionContainerVisible(), 'pre select show');
+
+          syn
+            .click(items[0], ()=>{
+
+              assert.equal(onSelectSpy.callCount, 1);
+
+              const selectedItem = onSelectSpy.args[0][0];
+
+              assert.propertyVal(selectedItem, 'title', 'tag:');
+              assert.propertyVal(selectedItem, 'explanation', 'search for annotations with a tag');
+
+              assert.isFalse(isSuggestionContainerVisible(), 'post select hide');
+
+              done();
+            });
+        });
+    });
+
+    var navigationExpectations = [
+      { travel: '[down]', selectedIndex: 0 },
+      { travel: '[up]', selectedIndex: 3 },
+      { travel: '[down][up]', selectedIndex: -1 },
+      { travel: '[up][down]', selectedIndex: -1 },
+      { travel: '[down][down][down][down]', selectedIndex: 3 },
+      { travel: '[down][down][down][down][down]', selectedIndex: -1 },
+      { travel: '[up][up][up][up]', selectedIndex: 0 },
+      { travel: '[up][up][up][up][up]', selectedIndex: -1 },
+      { travel: '[up][down][up][down][down][down][down][up]', selectedIndex: 1 },
+    ];
+
+    unroll('allows keyboard navigation', function(done, fixture){
+
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      const list = container.querySelector('.' + defaultConfig.classNames.list);
+
+      syn
+        .click(input)
+        .type(fixture.travel, () => {
+          var active = list.querySelectorAll('.' + defaultConfig.classNames.activeItem);
+          if(fixture.selectedIndex === -1){
+            assert.lengthOf(active, 0);
+          }else {
+            assert.isTrue(list.childNodes[fixture.selectedIndex].classList.contains(defaultConfig.classNames.activeItem));
+            assert.lengthOf(active, 1);
+          }
+          done();
+        });
+    }, navigationExpectations);
+
+
+    it('persists active navigation through list reduce', function(done){
+      new AutosuggestDropdownController(input, defaultConfig);
+
+      const list = container.querySelector('.' + defaultConfig.classNames.list);
+
+      syn
+        .click(input)
+        .type('[down][down]', () => {
+
+          assert.isTrue(list.childNodes[1].classList.contains(defaultConfig.classNames.activeItem));
+          assert.equal(list.childNodes[1].querySelector('.a-title').textContent.trim(),
+            defaultConfig.list[1].title);
+        })
+        .type('t', () => {
+
+          assert.isTrue(list.childNodes[0].classList.contains(defaultConfig.classNames.activeItem));
+          assert.equal(list.childNodes[0].querySelector('.a-title').textContent.trim(),
+            defaultConfig.list[1].title);
+
+          done();
+        });
+    });
+
+    it('can keyboard select', function(done){
```

Comment:
Description is perhaps a little terse - "Updates selection when keys are pressed"?